Source/swiftlint/Commands/LintCommand.swift

```diff
@@ -55,7 +55,10 @@ struct LintCommand: CommandType {
             violations += currentViolations
             reporter.reportViolations(currentViolations, realtimeCondition: true)
         }.flatMap { files in
-            reporter.reportViolations(violations, realtimeCondition: false)
+            if isWarningThresholdBroken(configuration, violations: violations) {
+                violations.append(createThresholdViolation(configuration.warningThreshold!))
+            }
+            reporter.reportViolations(violations, realtimeCondition: true)
```

Comment:
Ah, I know what's happening... this code should be this:

```swift
if isWarningThresholdBroken(configuration, violations: violations) {
  violations.append(createThresholdViolation(configuration.warningThreshold!))
  if reporter.isRealtime {
    reporter.reportViolations([violations.last!], realtimeCondition: true)
  }
}
reporter.reportViolations(violations, realtimeCondition: false)
```

Though this is pretty confusing and could stand to be refactored, but not in this PR.